### PR TITLE
Refactor pdf printer interface

### PR DESCRIPTION
**Pull request by Davidmattei**

### CODE REVIEW

File: Service/Pdf/PdfGenerator.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+{
+    /** @var PdfPrinterInterface */
+    private $pdfPrinter;
+
+    public function __construct(PdfPrinterInterface $pdfPrinter)
+    {
+        $this->pdfPrinter = $pdfPrinter;
+    }
+
+    public function getResponse(string $html): StreamedResponse
```
Comment (dameert) [inline Service/Pdf/PdfGenerator.php]:
```suggestion
    public function getStreamedResponse(string $html): StreamedResponse
```

File: Service/Pdf/PdfGenerator.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        $metaTags = $this->getMetaTags($html);
+        $pdfPrintOptions = new PdfPrintOptions($this->sanitizeMetaTags($metaTags));
+        $filename = $metaTags[PdfInterface::FILENAME] ?? 'export.pdf';
+
+        return $this->pdfPrinter->getStreamResponse(new Pdf($filename, $html), $pdfPrintOptions);
+    }
+
+    private function getMetaTags(string $html): array
```
Comment (dameert) [inline Service/Pdf/PdfGenerator.php]: Should be part of an HtmlParser class (S-OLID)

File: Service/Pdf/PdfGenerator.php (hunk 1 above)
Comment (Davidmattei) [reply Service/Pdf/PdfGenerator.php]: Fixed

File: Service/Pdf/PdfGenerator.php (hunk 2 above)
Comment (Davidmattei) [reply Service/Pdf/PdfGenerator.php]: Added HtmlCrawler but still need a getMetaTags function :) Logic is specific for PdfGenerator

File: Service/Pdf/PdfGenerator.php [hunk 3]
```diff
@@ -17,7 +17,7 @@ public function __construct(PdfPrinterInterface $pdfPrinter)
         $this->pdfPrinter = $pdfPrinter;
     }
 
-    public function getResponse(string $html): StreamedResponse
+    public function getStreamResponse(string $html): StreamedResponse
```
Comment (dameert) [inline Service/Pdf/PdfGenerator.php]:
```suggestion
    public function getStreamedResponse(string $html): StreamedResponse
```

**Pull request merged**